### PR TITLE
DPL Analysis: Use AddAt rather than Get+Set

### PR DESCRIPTION
**Pull request by ktf**
Should save at least one virtual method invokation (maybe 2 if final works as hoped).

### CODE REVIEW

**Comment by ktf** [comment]
When running under  IgProf on hyperloop this seems to remove completely the contribution of GetAt+SetAt. The bulk of the time is now in FindBin.

**Review by jgrosseo — APPROVED** [review]
Looks good to me. We can make a 1-1 comparison of the output to be sure

**Comment by ktf** [comment]
yes, please.

File: Framework/Core/include/Framework/StepTHn.h [hunk 1]
```diff
@@ -105,13 +113,40 @@ class StepTHnT : public StepTHn
...
+    } else if constexpr (std::is_same_v<TemplateArray, TArrayD>) {
+      auto* values = (TArrayD*)mValues[iStep];
+      values->AddAt(op.weight, op.bin);
+      if (mSumw2[iStep]) {
+        ((TArrayD*)mSumw2[iStep])->AddAt(op.weight, op.bin);
+      }
+    } else {
+      mValues[iStep]->SetAt(mValues[iStep]->GetAt(op.bin) + op.weight, op.bin);
+      if (mSumw2[iStep]) {
+        mSumw2[iStep]->SetAt(mSumw2[iStep]->GetAt(op.bin) + op.weight, op.bin);
```
Comment (ktf) [inline Framework/Core/include/Framework/StepTHn.h:138]: btw, is this actually correct?

**Pull request closed without merging**